crates/bifrost/src/error.rs

```diff
@@ -42,6 +42,8 @@ pub enum Error {
     AdminError(#[from] AdminError),
     #[error(transparent)]
     MetadataStoreError(#[from] restate_metadata_store::ReadWriteError),
+    #[error(transparent)]
+    ExtendLogChain(#[from] ExtendLogChainError),
```

Comment:
`ExtendLogChainError` include errors that overlap with the the parent enum `Error`. This is a sign that it's not a good error design.

Would it be possible to use `AdminError` for the extend-specific variants and use the top-level variants for the generic errors like UnknownLogId, basically following a similar strategy to what we do in `bifrost_admin.rs`'s `seal_and_auto_extend`?